sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -48,6 +48,29 @@
  */
 public class V2ExpressionSQLBuilder {
 
+  protected String escapeSpecialChar(String str) {
```

Comment:
Perhaps we should name this `escapeSpecialCharsForLikePattern` or similar, since this isn't a fully generalized escaping mechanism?